.ruff.toml

```diff
@@ -12,20 +12,45 @@ select = [
   "E",
   "F",
   "W",
-  #"UP",
-  #"PT"
+  "N",
+  "UP",
+  "BLE",
+  "A",
+  "C4",
+  "INP",
+  "PIE",
+  "T20",
+  "PT",
+  "RET",
+  "TID",
+  "PTH",
+  "PD",
+  "PLC",
+  "PLE",
+  "FLY",
+  "NPY",
+  "PERF",
+  "RUF",
 ]
 extend-ignore = [
   "E712",
   "E721",
-  # pycodestyle (E, W)
-  "E501", # LineTooLong                             # TODO! fix
+   # flake8-pie
+  "PIE808", # Disallow passing 0 as the first argument to range
```

Comment:
What prompted this rule?

Reply:
honestly don't know, it seems a weird thing to do so we're leaving it out

Reply:
To disallow 0 as a start input to range? Doesn't range default to 0?

Reply:
Yeah exactly, that seems like a thing anyone who writes in Python would expect

Reply:
> explicit is better than implicit

I don't know why you would have a linting rule which disallows being more explicit.

Reply:
Its the same as setting None as the second arg to get when its the default. Why?